### DISCUSSION TITLE
Add support for Rust-like in source tests.

### DISCUSSION THREAD

**Discussion by ChristianPavilonis** [pestphp/pest#894] [Ideas]
I drafted an example PR of how it could possibly work (#893).

I find that in the Rust world and while writing tests using Vitest writing them in-source is a better way to write unit tests not being divorced from the source code.

Example:
```php
<?php

class Greeter
{
    public static function greet(string $target = 'World'): string
    {
        return 'Hello, ' . $target . '!';
    }
}

// Write the test next to the class. is not run outside of running vendor/bin/pest
describe('Greeter', function () {
    it('greets the whole world by default', function () {
        expect(Greeter::greet())->toBe('Hello, World!');
    });

    it('greets whomever the target is when available', function() {
        expect(Greeter::greet("in-source tests"))->toBe('Hello, in-source tests!');
    });
});
```

The PR just conditionally declares Pest's functions with empty signatures, so when the file's are loaded those functions do get run but have no effect. I'm not sure what kind of overhead that might have on a large code-base though.

**Comment by paulshryock** [comment]
Deploying PHPUnit/Pest and all your tests to production seems like a bad idea.
